includes/HorizontalMirrorAnimation.hpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Memoris
+ * Copyright (C) 2015  Jean LELIEVRE
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/**
+ * @file HorizontalMirrorAnimation.hpp
+ * @brief plays the horizontal mirror animation
+ * @package animations
+ * @author Jean LELIEVRE <Jean.LELIEVRE@supinfo.com>
+ */
+
+#ifndef MEMORIS_ABSTRACTLEVELANIMATIONS_H_
+#define MEMORIS_ABSTRACTLEVELANIMATIONS_H_
+
+#include "AbstractLevelAnimation.hpp"
+
+namespace memoris
+{
+namespace animations
+{
+
+class HorizontalMirrorAnimation : public AbstractLevelAnimation
+{
+
+public:
+
+    /**
+     * @brief constructor, initializes the separator
+     *
+     * @param context shared pointer to the context to use
+     */
+    HorizontalMirrorAnimation(
+        const std::shared_ptr<utils::Context>& context
+    );
+
+    /**
+     * @brief renders the animation
+     *
+     * @param context shared pointer to the current context to use
+     * @param level shared pointer to the level to use
+     * @param floor the current floor to display in the animation
+     */
+    void render(
+        std::shared_ptr<utils::Context> context,
+        std::shared_ptr<entities::Level> level,
+        const unsigned short& floor
+    ) override;
+
+private:
+
+    /**
+     * @brief horizontally invert all the cells
+     */
+    void horizontallyInvertCells(
```

Comment:
according to what the function does, should be called replaceHighCellsByLowCells